src/azure-cli/azure/cli/command_modules/network/operations/dns.py

```diff
@@ -6,12 +6,46 @@
 # pylint: disable=line-too-long, protected-access, too-few-public-methods
 from knack.log import get_logger
 
-from azure.cli.core.aaz import has_value, register_command, AAZResourceIdArgFormat
-from ..aaz.latest.network.dns.record_set import Update as _RecordSetUpdate
+from azure.cli.core.aaz import has_value, register_command, AAZResourceIdArgFormat, AAZListArg, AAZResourceIdArg
+from ..aaz.latest.network.dns.record_set import Update as _RecordSetUpdate, Show as _RecordSetShow, \
+    ListByType as _RecordSetListByType, Delete as _RecordSetDelete, Create as _RecordSetCreate
+from ..aaz.latest.network.dns._list_references import ListReferences as _DNSListReferences
 
 logger = get_logger(__name__)
 
 
+@register_command("network dns list-references")
+class DNSListReferences(_DNSListReferences):
+    """ Returns the DNS records specified by the referencing targetResourceIds.
+
+    :example: Returns the DNS records specified by the referencing targetResourceIds.
+        az network dns list-references --parameters "/subscriptions/**921/resourceGroups/MyResourceGroup/providers/Microsoft.Network/trafficManagerProfiles/MyTrafficManager"
+    """
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.target_resources._registered = False
+
+        args_schema.parameters = AAZListArg(
+            options=["--parameters"],
+            help="A space-separated list of resource IDs for which referencing dns records need to be queried.",
+        )
+
+        parameters = args_schema.parameters
+        parameters.Element = AAZResourceIdArg()
+
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        parameters = args.parameters
+        if parameters is not None:
+            args.target_resources = []
+            for parameter in parameters:
+                args.target_resources.append({"id": parameter})
```

Comment:
Better to leverage `assign_aaz_list_arg`, e.g., https://github.com/Azure/azure-cli/blob/c44c9af9cb6b4e2b7cffeb343443860888b0fb1a/src/azure-cli/azure/cli/command_modules/network/custom.py#L1019-L1027.

Reply:
Used assign_aaz_list_arg